edx_oauth_client/middleware.py

```diff
@@ -66,70 +101,40 @@ def process_request(self, request):
-            'logout'
-        )
-
-        handle_local_urls += auth_process_urls + api_urls
-        is_auth = request.user.is_authenticated()
+        CUSTOM_OAUTH_PARAMS = getattr(settings, "CUSTOM_OAUTH_PARAMS", {})
 
-        if settings.DEBUG:
-            debug_handle_local_urls = ('debug', settings.STATIC_URL, settings.MEDIA_URL)
-            handle_local_urls += debug_handle_local_urls
-
-        if request.path == "/dashboard/" or request.path == "/dashboard":
-            if is_auth:
-                return redirect(os.path.join(OAUTH_PROVIDER_URL, 'members', request.user.username))
-            else:
-                return redirect(OAUTH_PROVIDER_URL)
-
-        r_url = re.compile(r'^/courses/(.*)/about').match(current_url)
-        if r_url:
-            return redirect(
-                os.path.join(os.path.join(OAUTH_PROVIDER_URL))
-            )
+        provider_url = CUSTOM_OAUTH_PARAMS.get("PROVIDER_URL", "")
+        dashboard_url_path = CUSTOM_OAUTH_PARAMS.get("DASHBOARD_URL_PATH")
+        courses_list_url_path = CUSTOM_OAUTH_PARAMS.get("COURSES_LIST_URL_PATH")
+        user_account_url_path = CUSTOM_OAUTH_PARAMS.get("USER_ACCOUNT_URL_PATH")
 
-        is_courses_list_or_about_page = False
-        r = re.compile(r'^/courses/%s/about' % settings.COURSE_ID_PATTERN)
-
-        if r.match(current_url):
-            is_courses_list_or_about_page = True
-
-        if request.path == "/courses/" or request.path == "/courses":
-            return redirect(os.path.join(OAUTH_PROVIDER_URL, 'courses'))
-
-        if request.path.startswith(
-                '/u/') or request.path == "/account/settings/" or request.path == "/account/settings":
-            if is_auth:
-                return redirect(os.path.join(OAUTH_PROVIDER_URL, 'members', request.user.username, 'profile'))
-            else:
-                return redirect(OAUTH_PROVIDER_URL)
-
-        if start_url not in handle_local_urls or is_courses_list_or_about_page:
-            if start_url.split('?')[0] not in handle_local_urls:
-                oauth_url = OAUTH_PROVIDER_URL.rstrip("/") + "/"
-                return redirect("%s%s" % (oauth_url, current_url))
+        current_url = request.get_full_path()
+        start_url_path, _, _ = urlparse(current_url).path.strip("/").partition("/")
 
-        if not is_auth and start_url not in auth_process_urls and \
-                        start_url not in api_urls:
-            request.session['force_auth'] = True
+        available_urls = API_URLS + LOCAL_URLS + OAUTH_PROCESS_URLS
+        if settings.DEBUG:
+            debug_handle_local_urls = ("debug", settings.STATIC_URL, settings.MEDIA_URL)
+            available_urls += debug_handle_local_urls
+
+        if request.user.is_authenticated():
+            if dashboard_url_path and request.path.strip("/") in reverse("dashboard"):
+                return redirect(urljoin(provider_url, dashboard_url_path))
```

Comment:
I would add new variable ala `result_url` and avoid returning values in every condition branch.
That would lead to better function stability and testing ability.

BtW does that method process_request is covered by the tests?